### PR TITLE
Fix #1353: make NRT-allocated memory visible to tracemalloc

### LINKED ISSUE #1353: Use CPython allocator in NRT
NRT should optionally use the CPython memory allocation functions (when imported from CPython). This would allow Numba-allocated memory to be seen by other utilities such as `sys.getallocatedblocks()`, `sys.debugmallocstats()`, and `tracemalloc`.

### PR DESCRIPTION
**Pull request by pitrou**
Also fixes an embarrassing typo in _pymodule.h.

### CODE REVIEW

**Comment by seibert** [comment]
I think if this PR merges in master now, the commit list will shrink substantially.

**Comment by pitrou** [comment]
Right :)

File: numba/runtime/nrt.h [hunk 1]
```diff
@@ -36,6 +36,11 @@ typedef int (*atomic_cas_func)(void * volatile *ptr, void *cmp, void *repl,
 typedef struct MemInfo MemInfo;
 typedef struct MemSys MemSys;
 
+typedef void *(*NRT_malloc_func)(size_t size);
+typedef void *(*NRT_realloc_func)(void *ptr, size_t new_size);
+typedef void (*NRT_free_func)(void *ptr);
```
Comment (seibert) [inline numba/runtime/nrt.h]: Is this function missing the initial `*`?

Comment (pitrou) [inline numba/runtime/nrt.h]: No, free() returns nothing.

Comment (seibert) [inline numba/runtime/nrt.h]: Oh right.  Blinded by '*'s!

**Pull request merged**